python/packages/autogen-core/tests/test_worker_runtime.py

```diff
@@ -300,3 +301,94 @@ class LoopbackAgentWithSubscription(LoopbackAgent): ...
     await worker.stop()
     await publisher.stop()
     await host.stop()
+
+
+@pytest.mark.asyncio
+async def test_duplicate_subscription() -> None:
+    host_address = "localhost:50059"
+    host = WorkerAgentRuntimeHost(address=host_address)
+    worker1 = WorkerAgentRuntime(host_address=host_address)
+    worker1_2 = WorkerAgentRuntime(host_address=host_address)
+    host.start()
+    try:
+        worker1.start()
+        await MyAgent.register(worker1, "worker1", lambda: MyAgent("worker1"))
+
+        worker1_2.start()
+
+        # Note: This passes because worker1 is still running
+        with pytest.raises(RuntimeError, match="Agent type worker1 already registered"):
+            await MyAgent.register(worker1_2, "worker1", lambda: MyAgent("worker1_2"))
+
+        # This is somehow covered in test_disconnected_agent as well as a stop will also disconnect the agent.
+        #  Will keep them both for now as we might replace the way we simulate a disconnect
+        await worker1.stop()
+
+        # Note: This is failing (doesn't raise the exeption). We get duplicate subscriptions
+        with pytest.raises(ValueError):
+            await MyAgent.register(worker1_2, "worker1", lambda: MyAgent("worker1_2"))
+
+    except Exception as ex:
+        raise ex
+    finally:
+        await worker1_2.stop()
+        await host.stop()
+
+
+@pytest.mark.asyncio
+async def test_disconnected_agent() -> None:
+    host_address = "localhost:50059"
+    host = WorkerAgentRuntimeHost(address=host_address)
+    host.start()
+    worker1 = WorkerAgentRuntime(host_address=host_address)
+    worker1_2 = WorkerAgentRuntime(host_address=host_address)
+
+    try:
+        worker1.start()
+        await MyAgent.register(worker1, "worker1", lambda: MyAgent("worker1"))
+
+        subscriptions1 = host.servicer.subscription_manager.subscriptions
```

Comment:
Can we use the private member for now and add a TODO to signal we will need to have a public API for subscriptions in the future.